### PR TITLE
Feature/permissions

### PR DESCRIPTION
**Pull request by Jaydev-byteam**
Refactored permissions to automatically ask for permission for HealthKit when user reaches Steps Page and Location information when user reaches Sleep Page. Additionally refactored Goals Screen to add snapshot listeners for the goals so the goal cards are dynamically updated with the user's sleep and step goals as they change on the specific goal pages.

### CODE REVIEW

File: src/container/GoalsScreenNav/GoalsScreenNav.js [hunk 1]
```diff
@@ -15,27 +14,24 @@ export default function GoalsScreenNav({navigation, user}) {
...
       </GoalsStack.Screen>
       <GoalsStack.Screen name="Steps">
         {props => <StepsScreen user={user} navigation={navigation} />}
       </GoalsStack.Screen>
-      <GoalsStack.Screen name="Sleep">
-        {props => <SleepScreen user={user} navigation={navigation} />}
-      </GoalsStack.Screen>
+      <GoalsStack.Screen
+        name="Sleep"
+        component={props => <SleepScreen user={user} navigation={navigation} />}
```
Comment (betteryou-ai) [inline src/container/GoalsScreenNav/GoalsScreenNav.js:30]: not sure if you're using the "user" object that you're passing in through navigation but you can simplify this component line by simply doing
`component={SleepScreen}` and it should automatically handle passing the props and navigation to the stack.

File: src/container/Router/Router.js [hunk 2]
```diff
@@ -55,7 +55,7 @@ export default function Router({isSignedIn}) {
                       }}>
                       <GoalsStack.Screen name="GoalsMain" component={GoalsScreenMain} />
                       <GoalsStack.Screen name="Steps" component={StepsScreen} />
-                      <GoalsStack.Screen name="Sleep" component={SleepScreen} />
+                      <GoalsStack.Screen name="Sleep" component={SleepScreen} options={{headerTintColor: '#fff'}}/>
```
Comment (betteryou-ai) [inline src/container/Router/Router.js:58]: I think we chatted about this last week, did the headerTintColor end up working out for you or is it still acting weird?

**Review by betteryou-ai — COMMENTED** [review]
Overall solid changes, I have a few suggestions we can meet about as well! 👍

File: src/container/GoalsScreenNav/GoalsScreenNav.js (hunk 1 above)
Comment (Jaydev-byteam) [reply src/container/GoalsScreenNav/GoalsScreenNav.js:30]: Sorry, I made changes to this file before I remembered that I had moved all the navigation content to Router.js In that file, the component is called simply as {sleepScreen}.

File: src/container/Router/Router.js (hunk 2 above)
Comment (Jaydev-byteam) [reply src/container/Router/Router.js:58]: Yes, headerTintColor was the correct solution, I just needed to call it correctly here.

**Pull request merged**